ca/ca_test.go

```diff
@@ -115,9 +115,11 @@ var (
 	OIDExtensionCTPoison = asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 11129, 2, 4, 3}
 
 	issuanceModes = []IssuanceMode{
-		{name: "non-precertificate", usePrecertificateFlow: false, enablePrecertificateFlow: false},
-		{name: "precertificate", usePrecertificateFlow: true, enablePrecertificateFlow: true},
-		{name: "precertificate-disabled", usePrecertificateFlow: true, enablePrecertificateFlow: false},
+		{name: "non-precertificate", issuePrecertificate: false, issueCertificateForPrecertificate: false, enablePrecertificateFlow: false},
+		{name: "precertificate", issuePrecertificate: true, issueCertificateForPrecertificate: false, enablePrecertificateFlow: true},
+		{name: "precertificate-disabled", issuePrecertificate: true, issueCertificateForPrecertificate: false, enablePrecertificateFlow: false},
+		{name: "certificate-for-precertificate", issuePrecertificate: false, issueCertificateForPrecertificate: true, enablePrecertificateFlow: true},
+		{name: "certificate-for-precertificate-disabled", issuePrecertificate: false, issueCertificateForPrecertificate: true, enablePrecertificateFlow: false},
```

Comment:
Do you think there's value in a test case that checks the mutually exclusive values being set? e.g.
`{issuePrecertificate: true, issueCertificateForPrecertificate: true, enablePrecertificateFlow: true},` I suspect the answer is "no" but I think it's the only omission here and I wanted to check explicitly.